std/assembly/array.ts

```diff
@@ -25,6 +25,30 @@ export class Array<T> {
     this.__capacity = this.__length = capacity;
   }
 
+  every(callbackfn: (element: T, index: i32, array: Array<T>) => bool): bool {
+    var toIndex: i32 = this.__length;
+    var i: i32 = 0;
+    while (i < toIndex && i < this.__length) {
```

Comment:
Double checking the same value. It seems you can do simpler:

```ts
var len: i32 = this.__length;
var i = 0;
while (i < len) {
   ...
}
```
```

Reply:
Actually i can't. Array size can be reduced during iteration, you can call `.pop()` method inside of callback. So array length will be decreased during iteration, and you will iterate through items which are not exists anymore. Of course it's a terrible practice, side effect etc, but this check is required by spec.

Reply:
Ok, I see. Makes sense. Anyway this will simpler. Just:

```ts
while (i < this.__length) { ... }
```
Early returning with cached ```i < toIndex``` happened in very rare cases but in all other cases produce overhead.

Reply:
And not correct if we make push during iteration. cached toIndex trigger early return, while actual __length increased and still return **true**

Reply:
Added elements shouldn't affect iteration also. Just check this in your browser console:
```javascript
[1, 2, 3].reduce((a, b, i, arr) => {
  arr.push(1);
  return a + b;
}) === 6
```
Same works for any of `Array.prototype` methods and also required by spec. We need to both cache length and check it on each iteration.

Reply:
Right. Very strange. So we can reduce elements during iteration but not increase.

Reply:
I also haven't noticed this behaviour before. So, this project helps us to learn edge-cases of js since we should check spec every time :D